tools/clang/include/clang/Sema/SemaHLSL.h

```diff
@@ -24,6 +24,19 @@
 #include "clang/Sema/Template.h"
 #include "clang/Sema/TemplateDeduction.h"
 
+#include <optional>
+extern std::optional<bool> disableHLSLIntrinsicsGlobalVariableBecauseIDontCare;
+
+inline bool shouldDisableHLSLIntrinsics()
+{
+  assert(disableHLSLIntrinsicsGlobalVariableBecauseIDontCare.has_value());
+
+  if (!disableHLSLIntrinsicsGlobalVariableBecauseIDontCare.has_value())
+    exit(-1);
+
+  return disableHLSLIntrinsicsGlobalVariableBecauseIDontCare.value();
+}
```

Comment:
hide the global var (they mess with DLL delay loads), make the function non-inline and export like the `hlsl::Check` functions.